src/Blocks/BlockHandler.h

```diff
@@ -123,7 +123,14 @@ class cBlockHandler
 	/** Returns if this block drops if it gets destroyed by an unsuitable situation.
 	Default: true */
 	virtual bool DoesDropOnUnsuitable(void);
-	
+
+	/** Returns the block's height */
+	virtual double GetBlockHeight(void);
+
+	/** Tests if a_Position is inside the block where a_Position is always [0..1[
+	Note that this is considered from a "top-down" perspective i.e. a top-half slab will always return true */
+	virtual bool IsInsideBlock(const Vector3d & a_Position, const NIBBLETYPE & a_BlockMeta);
```

Comment:
Fences are unaccounted for, I admit to that much, but that can be solved by simply sampling the block one block lower, but leaving the sampling point unchanged. That'd make the range [0..1,0..2,0..1[
Notably this can basically extend to any sampled blocks in the vicinity if there are ever blocks that have a single anchor but expand to multiple physical blocks.